ssl/t1_lib.c

```diff
@@ -644,7 +649,7 @@ int tls1_check_ec_tmp_key(SSL *s, unsigned long cid)
             return 0;
         curve_id[0] = 0;
         /* Check this curve is acceptable */
-        if (!tls1_check_ec_key(s, curve_id, NULL))
+        if (!tls1_check_ec_key(s, curve_id, NULL, 1))
```

Comment:
Just checking that here we're selecting an ECDHE group, not a certificate, and so MUST of course match the client curve list.  Right?

Reply:
This is in the context of suite B. We are considering selecting an ECDHE based ciphersuite and want to check if it is an acceptable choice. Suite B mandates specific curves to be used for ECDHE for a given ciphersuite. Here we are checking that the group associated with the ciphersuite we are considering is acceptable to us and the client.